common/djangoapps/config_models/admin.py

```diff
@@ -178,8 +182,38 @@ def add_view(self, request, form_url='', extra_context=None):
             get = request.GET.copy()
             source_id = int(get.pop('source')[0])
             source = get_object_or_404(self.model, pk=source_id)
-            get.update(models.model_to_dict(source))
+            source_dict = models.model_to_dict(source)
+
+            # read FileField content into request.FILES for each field, if:
+            # * field is a FileField
+            # * user hasn't ticked the "clear" checkbox
+            # * user hasn't uploaded a new file
+            # * source object has a file that can be read
+            for field_name, field_value in source_dict.items():
+                field = source._meta.get_field(field_name)
+                form_field = field.formfield()
+                clear_checkbox = False
+                if form_field and hasattr(form_field, 'widget') and hasattr(form_field.widget, 'clear_checkbox_name'):
+                    clear_checkbox = CheckboxInput().value_from_datadict(
+                        request.POST, request.FILES, form_field.widget.clear_checkbox_name(field_name))
+
+                if (isinstance(field, FileField) and
+                        not clear_checkbox and
+                        (field_name not in request.FILES) and
+                        (field_value is not None)):
+                    try:
+                        file_content = default_storage.open(field_value)
+                        request.FILES[field_name] = SimpleUploadedFile.from_dict({
+                            'filename': unicode(field_value),
+                            'content': file_content.read()
```

Comment:
Not a big fan of this approach... If the file is hosted on S3, this approach would require that the LMS download the file and upload it again, which wastes time and bandwidth in order to accomplish nothing.

Is there not some way that we can just say, "if the user has not chosen to clear or replace the file, set the string value of the file path stored in the database on the new model to be the same as the string value on the source model."

i.e. In the database, django model FileFields are simply represented as a string path, e.g. "some-logo.png". All we really need to do is to bypass the file validation and uploading logic to set the raw value of the new field to the same string as the source model, i.e. "some-logo.png".